src/common/ui/TabBar/TabBar.tsx

```diff
@@ -5,27 +5,36 @@ import { Tab } from 'common/ui';
 import styles from './TabBar.module.scss';
 
 interface TabBarProps {
-  tabsLabels: string[]
+  tabs: string[];
+  defaultActiveTab: string;
+  onActiveTabChanged: (tabLabel: string) => void;
 }
 
-function TabBarComponent({ tabsLabels }: TabBarProps) {
-  const [activeTabIndex, setActiveTabIndex] = useState(0);
+const getDefaultActiveTabIndex = (tabs: string[], defaultActiveTabLabel: string) => {
+  return tabs.findIndex((item) => item === defaultActiveTabLabel);
+};
 
-  const handleTabClick = useCallback(memoize((tabIndex: number) => {
+function TabBarComponent({ tabs, defaultActiveTab, onActiveTabChanged }: TabBarProps) {
+  const [activeTabIndex, setActiveTabIndex] = useState(
+    getDefaultActiveTabIndex(tabs, defaultActiveTab),
+  );
+
+  const handleTabClick = useCallback(memoize((tabIndex: number, tabLabel: string) => {
```

Comment:
Here we go again.

Create 2 functions wrapped in `useCallback` and call one inside another.